resources/providers/awslib/config_provider.go

```diff
@@ -21,28 +21,26 @@ import (
 	"context"
 	awssdk "github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/elastic/beats/v7/x-pack/libbeat/common/aws"
-	"github.com/elastic/elastic-agent-libs/logp"
 )
 
 type ConfigProvider struct {
 	MetadataProvider MetadataProvider
 }
+
 type ConfigProviderAPI interface {
-	InitializeAWSConfig(ctx context.Context, cfg aws.ConfigAWS, log *logp.Logger) (awssdk.Config, error)
+	InitializeAWSConfig(ctx context.Context, cfg aws.ConfigAWS) (awssdk.Config, error)
 }
 
-func (p ConfigProvider) InitializeAWSConfig(ctx context.Context, cfg aws.ConfigAWS, log *logp.Logger) (awssdk.Config, error) {
+func (p ConfigProvider) InitializeAWSConfig(ctx context.Context, cfg aws.ConfigAWS) (awssdk.Config, error) {
 	awsConfig, err := aws.InitializeAWSConfig(cfg)
 	if err != nil {
 		return awssdk.Config{}, err
 	}
-
 	metadata, err := p.MetadataProvider.GetMetadata(ctx, awsConfig)
 	if err != nil {
-		log.Errorf("MetadataProvider.GetMetadata Error: %v, setting AWSConfig region to default - %s", err, awsConfig.Region)
-		return awsConfig, nil
+		return awssdk.Config{}, err
```

Comment:
reverted to pre cspm changes.

Reply:
Don't we want to keep the log though?

Reply:
Also, I think we should return a pointer to `awssdk.Config` and that way we will be able to return `nil` here.

Reply:
Regarding keeping the log, there is no real point in logging and returning an error.
This error is propagated all the way to `cloudbeat.go` and will cause the beat to stop.